sn_networking/src/record_store.rs

```diff
@@ -202,71 +185,86 @@ impl DiskBackedRecordStore {
     }
 
     /// Prune the records in the store to ensure that we free up space
-    /// for new records.
-    ///
-    /// This can happen if we're full, or once in every X PUTs
+    /// for the incoming record.
     ///
     /// An error is returned if we prune and we're _still_ at max_record count
-    fn prune_storage_if_needed(&mut self) -> Result<()> {
+    fn prune_storage_if_needed_for_record(&mut self, r: &Key) -> Result<()> {
         let num_records = self.records.len();
-        let mut should_prune = num_records >= self.config.max_records;
-        if !should_prune {
-            // check the modulo of record count over 10, if 0 we should prune
-            should_prune = num_records % PRUNE_ONCE_EVERY_PUTS == 0;
-        }
 
-        if should_prune {
-            self.prune_storage();
-            let new_num_records: usize = self.records.len();
-            trace!("A pruning reduced number of record in hold from {num_records} to {new_num_records}");
-            if new_num_records >= self.config.max_records {
-                warn!("Record not stored. Maximum number of records reached. Current num_records: {num_records}");
-                return Err(Error::MaxRecords);
+        // sort records by distance to our local key
+        let mut records = self.records.iter().cloned().collect::<Vec<_>>();
+        records.sort_by_key(|k| {
```

Comment:
According to the Rust docs on `sort_by_key`, we can use the 'unstable' sort variant:

When applicable, unstable sorting is preferred because it is generally faster than stable sorting and it doesn’t allocate auxiliary memory. See [sort_unstable_by_key](https://doc.rust-lang.org/std/primitive.slice.html#method.sort_unstable_by_key).